### PR TITLE
CS-Script update v1.7.29

### PR DESCRIPTION
**Pull request by oleg-shilo**
CS-Script update v1.7.29

### CODE REVIEW

**Comment by donho** [comment]
@oleg-shilo 
I suppose CS-Script v1.7.29 x64 works only with Notepad++ x64 from v8.3 to v8.3.3 ?
Which version is the last one to work with Notepad++ x64 v8.2.1 but not v8.3.x ?

**Comment by oleg-shilo** [comment]
@donho 

CS-Script v1.7.29 x64 works in default mode with Notepad++ x64 from v8.3 and higher. 
It will also work in legacy API mode on early versions of npp.

I have recalibrated the npp-API (legacy_API <-> default_API) auto-switch algorithm to be triggered by v8.3.

When the compatibility setting in `pl.x64.json` is available I will start using it for CS-Script releases. But for now, resorted to dynamic switching to the appropriate API interface.

The problem was with the CS-Script v1.7.28 x64

Working on npp prior v8.3 
Not working on npp v8.3 - v8.3.1 
Working on npp v8.3.1 and higher

**Comment by oleg-shilo** [comment]
@donho 

The explanation above is just to give you the context. But as it goes to answering your very question:

**Last plugin version to work on npp v8.21 - cs-script v1.7.26.0 (commit: 9c4cedc6c23d62d29459a46321c2ae99de449619) 
CS-Script v1.7.29 will work from v8.3.0 and newer.**

I am not sure if I should use the compatibility feature as you wrote "the plugin compatibility feature will only apply to Notepad++ v8.3.3 and versions superior to v8.3.3.".

**Comment by donho** [comment]
@oleg-shilo 

Thank you for your detailed reply.

> I am not sure if I should use the compatibility feature as you wrote "the plugin compatibility feature will only apply to Notepad++ v8.3.3 and versions superior to v8.3.3.".

You should add (just after `"version": "1.7.29"`) `"npp-compatible-versions": "[8.3,]"` and `"old-versions-compatibility": "[;1.7.26][,8.2.1]`

It means v1.7.29 is compatible with all Npp version from v8.3, and all version of CS-Script to v1.7.26 are compatible with Notepad++ all versions to v8.2.1.

This plugin list will be included in Notepad++ v8.3.3, so if user has CS-Script v1.7.26 with Notepad++ v8.2.1, and want to update to Notepad++ v8.3.3, v1.7.26 will be filtered and not loaded, therefore no crash.

**Comment by oleg-shilo** [comment]
Done: [ed05446](https://github.com/notepad-plus-plus/nppPluginList/pull/425/commits/ed0544695930fb75eac3b4eca97a6e4505f0c057)

Txs

File: src/pl.x64.json [hunk 1]
```diff
@@ -192,9 +192,11 @@
 		{
 			"folder-name": "CSScriptNpp",
 			"display-name": "CS-Script - C# Intellisense",
-			"version": "1.7.28.0",
-			"id": "67bd792a292363a5b32e13b5ab6eba2495c964c8e9c5bf35a7393886a5b18d97",
-			"repository": "https://github.com/oleg-shilo/cs-script.npp/releases/download/v1.7.28.0/CSScriptNpp.1.7.28.0.x64.zip",
+			"version": "1.7.29.0",
+			"npp-compatible-versions": "[8.3,]",
+			"old-versions-compatibility": "[;1.7.26][,8.2.1]",
```
Comment (donho) [inline src/pl.x64.json:197]: It should be `[,1.7.26][,8.2.1]`, not `[;1.7.26][,8.2.1]`

**Review by donho — CHANGES_REQUESTED** [review]
please use `[,1.7.26][,8.2.1]` instead of `[;1.7.26][,8.2.1]`

**Comment by oleg-shilo** [comment]
I see. I just copied your code from the post. Done. Fixed.

Can you please publish the specification for that. When it is ready

**Comment by donho** [comment]
> I see. I just copied your code from the post.

My bad. Sorry for that.

> Can you please publish the specification for that. When it is ready

https://github.com/notepad-plus-plus/npp-usermanual/pull/327

**Pull request merged**